src/devices/Ft232H/Ft232HI2c.cs

```diff
@@ -47,9 +47,15 @@ public override void WriteRead(ReadOnlySpan<byte> writeBuffer, Span<byte> readBu
         /// <inheritdoc/>
         protected override void Dispose(bool disposing)
         {
-            _i2cBus?.RemoveDevice(_deviceAddress);
-            _i2cBus = null!;
-            _settings = null!;
```

Comment:
The reason this is set to null is so that GC can collect this. How could this code be causing issue? Is it possible the problem is actually related to this: https://github.com/dotnet/iot/pull/2032#discussion_r1100084654 ?

Reply:
OK, so the I2cBus should not then be set to null but settings still should.
Seems that there is some edge problem. not setting to null won't change much. If the I2cBus is not reused later, it will just be GC as well.

So then fix should more looks like:

```csharp
protected override void Dispose(bool disposing)
{
    try
    {
        _i2cBus?.RemoveDevice(_deviceAddress);
    }
    catch (ArgumentException)
    {
        // dont worry if the device has already been removed
    }

    _settings = null!;
    base.Dispose(disposing);
}
```

Reply:
@swharden thanks a lot for the deeper investigation.

Reply:
Shouldn't there be a check for if(disposing) so that this doesn't attempt to use managed objects that may already have been GC'd during finalization?

Reply:
I made the change suggested by @Ellerbach in https://github.com/dotnet/iot/pull/2032#discussion_r1101147901 and am pushing updates to this PR now